colossalai/shardformer/layer/_operation.py

```diff
@@ -451,14 +743,20 @@ class _SplitForwardGatherBackward(torch.autograd.Function):
     """
 
     @staticmethod
-    def forward(ctx, input_, dim, process_group):
+    def forward(ctx, input_, dim, process_group, grad_scale=None):
         ctx.process_group = process_group
         ctx.dim = dim
+        ctx.grad_scale = grad_scale
         return _split(input_, dim, process_group)
 
     @staticmethod
     def backward(ctx, grad_output):
-        return _gather(grad_output, ctx.dim, ctx.process_group), None, None
+        if ctx.grad_scale is not None:
+            if ctx.grad_scale == "up":
```

Comment:
This is not intuitive. A float value may be better. E.g. `dist.get_world_size(ctx.process_group)` rather than "up", and `1 / dist.get_world_size(ctx.process_group)` rather than "down"